modules/oadp-ssec-encrypted-backups.adoc

```diff
@@ -0,0 +1,162 @@
++
+[source,terminal]
+----
+$ ln -s sse_encoded.key customer-key
+----
+
+. Create an {product-title} secret:
+** If you are initially installing and configuring {oadp-short}, you can create the AWS credential and encryption key secret at the same time by running the following command:
++
+[source,terminal]
+----
+$ oc create secret generic cloud-credentials --namespace openshift-adp --from-file cloud=<path>/openshift_aws_credentials,customer-key=<path>/sse_encoded.key
+----
+** If you are updating an existing installation, edit the values of the `cloud-credential` `secret` block of the `DataProtectionApplication` CR manifest, as in the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+data:
+  cloud: W2Rfa2V5X2lkPSJBS0lBVkJRWUIyRkQ0TlFHRFFPQiIKYXdzX3NlY3JldF9hY2Nlc3Nfa2V5P<snip>rUE1mNWVSbTN5K2FpeWhUTUQyQk1WZHBOIgo=
+  customer-key: v+<snip>TFIiq6aaXPbj8dhos=
+kind: Secret
+# ...
+----
+. Edit the value of the `customerKeyEncryptionFile` attribute in the `backupLocations` block of the `DataProtectionApplication` CR manifest, as in the following example:
++
+[source,yaml]
+----
+spec:
+  backupLocations:
+    - velero:
+        config:
+          customerKeyEncryptionFile: /credentials/customer-key
+          profile: default
+# ...
+----
++
+[WARNING]
+====
+You must restart the Velero pod to remount the secret credentials properly on an existing installation.
+====
++
+The installation is complete, and you can back up and restore {product-title} resources. The data saved in AWS S3 storage is encrypted with the new key, and you cannot download it from the AWS S3 console or API without the additional encryption key.
+
+.Verification
+
+To verify that you cannot download the encrypted files without the inclusion of an additional key, create a test file, upload it, and then try to download it.
+
+. Create a test file by running the following command:
++
+[source,terminal]
+----
+$ echo "encrypt me please" > test.txt
+----
+. Upload the test file by running the following command:
++
+[source,terminal]
+----
+$ aws s3api put-object \
+  --bucket <bucket> \
+  --key test.txt \
+  --body test.txt \
+  --sse-customer-key fileb://sse.key \
+  --sse-customer-algorithm AES256
+----
+.  Try to download the file. In either the Amazon web console or the terminal, run the following command:
++
+[source,terminal]
+----
+$ s3cmd get s3://<bucket>/test.txt test.txt
+----
++
+The download fails because the file is encrypted with an additional key.
+
+. Download the file with the additional encryption key by running the following command:
++
+[source,terminal]
+----
+$ aws s3api get-object \
+    --bucket <bucket> \
+    --key test.txt \
+    --sse-customer-key fileb://sse.key \
+    --sse-customer-algorithm AES256 \
+    downloaded.txt
+----
+
+. Read the file contents by running the following command:
++
+[source,terminal]
+----
+$ cat downloaded.txt
+encrypt me please
```

Comment:
Oh, missed this. Since "encrypt me please" is really example output, that should be its own code block entitled "Example output".